firebase-installations/src/main/java/com/google/firebase/installations/CrossProcessLock.java

```diff
@@ -25,67 +26,40 @@
 /** Use file locking to acquire a lock that will also block other processes. */
 class CrossProcessLock {
   private static final String TAG = "CrossProcessLock";
-  private final FileChannel channel;
-  private final FileLock lock;
 
-  private CrossProcessLock(FileChannel channel, FileLock lock) {
-    this.channel = channel;
-    this.lock = lock;
+  interface CrossProcessPersistedInstallation {
+    PersistedInstallationEntry getPersistedInstallations();
   }
 
   /**
-   * Create and return a lock that is exclusive across processes. If another process has the lock
-   * then this call will block until it is released.
+   * Create a lock that is exclusive across processes. If another process has the lock then this
+   * call will block until it is released.
    *
    * @param lockName the lockname is global across all processes of an app
    * @return a CrossProcessLock if success. If the lock failed to acquire (maybe due to disk full or
    *     other unexpected and unsupported permissions) then null will be returned.
    */
-  static CrossProcessLock acquire(Context appContext, String lockName) {
-    FileChannel channel = null;
-    FileLock lock = null;
-    try {
-      File file = new File(appContext.getFilesDir(), lockName);
-      channel = new RandomAccessFile(file, "rw").getChannel();
-      // Use the file channel to create a lock on the file.
-      // This method blocks until it can retrieve the lock.
-      lock = channel.lock();
-      return new CrossProcessLock(channel, lock);
+  static PersistedInstallationEntry whileLocked(
+      Context applicationContext,
+      String lockName,
+      CrossProcessPersistedInstallation persistedInstallation) {
+
+    File file = new File(applicationContext.getFilesDir(), lockName);
+    try (RandomAccessFile randomAccessFile = new RandomAccessFile(file, "rw");
+        FileChannel channel = randomAccessFile.getChannel();
+        // Use the file channel to create a lock on the file.
+        // This method blocks until it can retrieve the lock.
+        FileLock lock = channel.lock()) {
```

Comment:
Remind me again please:
Why can we use `try(<resource>)` here and couldn't use it before?

Reply:
We couldn't use try-with-resources before because there was an issue when migrating it to google3. That seems to be fixed now.